source/Octopus.Client/LocationChecker.cs

```diff
@@ -0,0 +1,42 @@
+using System;
+using System.Diagnostics;
+using System.Linq;
+using System.Reflection;
+using Microsoft.Win32;
+using Octopus.Client.Logging;
+
+namespace Octopus.Client
+{
+#if FULL_FRAMEWORK
+    class LocationChecker
+    {
+        private static readonly ILog Logger = LogProvider.For<LocationChecker>();
+
+        public static void CheckAssemblyLocation()
+        {
+            var octopusRegNode = Registry.LocalMachine.OpenSubKey(@"Software\Octopus");
+            CheckBasedOnApplication(octopusRegNode, "Server");
+            CheckBasedOnApplication(octopusRegNode, "Tentacle");
+        }
+
+        private static void CheckBasedOnApplication(RegistryKey octopusRegNode, string application)
+        {
+            var applicationRegNode = octopusRegNode.OpenSubKey(application);
+            if (applicationRegNode != null)
+            {
+                var installationFolder = (string) applicationRegNode.GetValue("InstallLocation");
+                var currentAssemblyLocation = typeof(LocationChecker).Assembly.Location;
+
+                if (currentAssemblyLocation.Contains(installationFolder))
+                {
+                    var warningMessage = $"Using Octopus.Client from the {application}'s installation folder has been deprecated. In future versions it may not be shipped with {application}. Please see https://g.octopushq.com/usingoctopusclientfrominstallationfolder.";
```

Comment:
"It seems your script is referencing Octopus.Client.dll from the {application}'s installation folder. This is a heads up to let you know that in future releases Octopus.Client.dll will no longer be shipped with {application}. Please refer to https://g.octopushq.com/usingoctopusclientfrominstallationfolder for instructions on how to address this issue.